EIPS/eip-5994.md

```diff
@@ -0,0 +1,85 @@
+
+## Motivation
+
+This Token Pods concept allows to have permissionless extensions for tokens, which can be used for incentivisations (aka farms) or delegations or any other ideas.
+
+Example implementation can be found here:
+- https://github.com/1inch/erc20-pods
+
+Examples of Pods:
+- https://github.com/1inch/farming
+- https://github.com/1inch/delegating
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+```solidity
+interface IPod {
+    function updateBalances(address from, address to, uint256 amount) external;
+}
+
+interface IERC20Pods is IERC20 {
+    function hasPod(address account, address pod) external view returns(bool);
+    function podsCount(address account) external view returns(uint256);
+    function podAt(address account, uint256 index) external view returns(address);
+    function pods(address account) external view returns(address[] memory);
+    function podBalanceOf(address pod, address account) external view returns(uint256);
+
+    function addPod(address pod) external;
+    function removePod(address pod) external;
+    function removeAllPods() external;
+}
+
+interface IERC721Pods is IERC721 {
+    function hasPod(address account, address pod) external view returns(bool);
+    function podsCount(address account) external view returns(uint256);
+    function podAt(address account, uint256 index) external view returns(address);
+    function pods(address account) external view returns(address[] memory);
+    function podBalanceOf(address pod, address account) external view returns(uint256);
+
+    function addPod(address pod) external;
+    function removePod(address pod) external;
+    function removeAllPods() external;
+}
+```
+
+### Pod specification
+
+- Pod MUST implement interface `IPod` and fit execution of `updateBalances` method in 200,000 gas and never revert.
+
+- Pod SHOULD make sure `updateBalances(...)` method was triggered by `token` it was designated for.
+
+- Pod can track changes of balances of users who voluntarily added this pod, all other account will be represented as `address(0)` and interactions with such accounts could be considered as mints and burns.
+
+- Pod SHOULD NOT try to change balances of the token it was designated for, since it would introduce reentrancy, which will not pass.
+
+## Rationale
+
+## Backwards Compatibility
+
+Fully backward compatibile with ERC20 and ERC721.
+
+## Security Considerations
+
+Pods implementations will never block balance update, but can eat upto 200,000 gas. Number of added Pods per account is limited to some immutable value (reasonable amount taking into account max transaction gas limit).
```

Comment:
I am a little worry about the introduction of Pods and its hooks will massively increase the chance of renetry attack, could you provide your considerations about this?